packages/local_auth/local_auth_darwin/darwin/Classes/FLALocalAuthPlugin.m

```diff
@@ -153,6 +153,26 @@ - (void)showAlertWithMessage:(NSString *)message
           dismissButtonTitle:(NSString *)dismissButtonTitle
      openSettingsButtonTitle:(NSString *)openSettingsButtonTitle
                   completion:(FLADAuthCompletion)completion {
+#if TARGET_OS_OSX
+  NSAlert *alert = [[NSAlert alloc] init];
+  [alert setMessageText:message];
+  [alert addButtonWithTitle:dismissButtonTitle];
+  if (openSettingsButtonTitle != nil) {
+    [alert addButtonWithTitle:openSettingsButtonTitle];
+  }
+  [alert beginSheetModalForWindow:NSApp.keyWindow
+                completionHandler:^(NSModalResponse returnCode) {
+                  if (returnCode == NSAlertSecondButtonReturn) {
+                    NSURL *url = [NSURL URLWithString:@"x-apple.systempreferences:com.apple."
```

Comment:
https://github.com/flutter/packages/pull/5766/files#r1486380427

Reply:
https://gist.github.com/rmcdongit/f66ff91e0dad78d4d6346a75ded4b751

Reply:
You've marked this as resolved, but I don't see a code comment in the current version, as requested by the linked review.

More importantly though, what the link you provided appears to be describing is a reverse-engineered system that uses private keys and has a proven track record of fragility to OS updates. That means that:
- there's no guarantee that Apple would consider this app-store compliant, and
- it is very likely to break without warning in the future, leaving a button that doesn't actually work.

Am I misunderstanding the discussion there? If not, then this needs a different solution for macOS that is more robust even if it's not as featureful.

Reply:
What do you suggest then in order to open the settings app? Here is another source with all the url schemes for settings: https://www.mbsplugins.de/archive/2020-04-05/MacOS_System_Preference_Links